// ycb/uh/tdv2-applet-uh/configs/assets.js
// WARNING: This file is only used for development and functional testing, this will not be used by the application.

var yahoocss = require('yahoocss');

module.exports = [
{
    settings: ['master'],
    blob: [
        {
            defer: true,
            location: 'bottom',
            type: 'js',
            value: [
                'if (window.location.search.indexOf("config.uh.ui.theme=dark") > -1) {',
                'document.querySelector("#app .UH").style.backgroundColor="black";',
                '}'
            ].join(' ')
        }
    ],
    css: [
        {
            location: 'top',
            value: '/css/atomic.css'
        },
        {
            location: 'top',
            value: yahoocss.getCDN('bundle.css')
        },
        {
            location: 'top',
            value: yahoocss.getCDN('patch.css')
        }
    ],
    js: [
        {   location: 'top',
            value: 'https://s.yimg.com/ss/rapid-3.31.js'
        },
        {
            location: 'bottom',
            value: '/js/main.js'
        }
    ]
},
{
    settings: ['device:desktop-lite'],
    blob: [],
    js: [
    ]
},
{
    settings: ['device:smartphone'],
    blob: [
        {
//            location: 'top',
//            type: 'css',
//            value:  fs.readFileSync(__dirname + '/../public/css/mobile_header.css', 'utf-8')
        }
    ]
}];
