// ycb/uh/tdv2-applet-uh/configs/services/theme/site/beauty.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:beauty', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_beauty_en-US_s_f_pw_351x40_beauty.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_beauty_en-US_s_f_pw_351x40_beauty_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:beauty', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_beauty_en-US_s_f_pw_351x40_beauty.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_beauty_en-US_s_f_pw_351x40_beauty_2x.png',
                            partner: {
                                content_2: 'W(85px)'
                            },
                            yahoo: {
                                content_2: 'Mstart(85px)!'
                            }
                        }
                    }
                }
            }
        },
        {
            settings: ['site:beauty', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_beauty_en-US_s_f_pw_351x40_beauty.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_beauty_en-US_s_f_pw_351x40_beauty_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:beauty', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_beauty_en-US_s_f_pw_351x40_beauty.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_beauty_en-US_s_f_pw_351x40_beauty_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
