// ycb/uh/tdv2-applet-uh/configs/services/theme/site/katiecouric.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:katiecouric', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-US_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:katiecouric', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:katiecouric', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:katiecouric', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
