// ycb/uh/tdv2-applet-uh/configs/services/theme/site/movies.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:movies', 'lang:de-DE'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_de-DE_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_de-DE_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-AE_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-AE_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-CA'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-CA_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-CA_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-GB'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-GB_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-GB_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-IN'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-IN_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-IN_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-SG'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-SG_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-SG_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_en-US_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_en-US_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:es-ES'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_es-ES_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_es-ES_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:es-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_es-US_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_es-US_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:fr-FR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_fr-FR_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_fr-FR_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:it-IT'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_it-IT_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_it-IT_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'lang:pt-BR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_movies_pt-BR_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_movies_pt-BR_s_f_pw_351x40_movies_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:movies', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_movies_en-US_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_movies_en-US_s_f_pw_351x40_movies_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_movies_en-US_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_movies_en-US_s_f_pw_351x40_movies_2x.png'
                        }
                    }
                }
            }
        },
        // Remark: LOGO is not ready.
        // {
        //     settings: ['site:movies', 'partner:rogers', 'lang:fr-CA'],
        //     uh: {
        //         logo: {
        //             style: {
        //                 cobrand: {
        //                    image_1x: 'https://s.yimg.com/rz/d/rogers_movies_fr-CA_s_f_pw_351x40_movies.png',
        //                    image_2x: 'https://s.yimg.com/rz/d/rogers_movies_fr-CA_s_f_pw_351x40_movies_2x.png'
        //                 }
        //             }
        //         }
        //     }
        // },
        {
            settings: ['site:movies', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_movies_en-US_s_f_pw_351x40_movies.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_movies_en-US_s_f_pw_351x40_movies_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
