// ycb/uh/tdv2-applet-uh/configs/services/theme/site/news.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:news', 'lang:de-DE'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_de-DE_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_de-DE_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-AE_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-AE_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:en-CA'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-CA_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-CA_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:en-GB'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-GB_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-GB_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:en-IN'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-IN_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-IN_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:en-SG'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-SG_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-SG_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        }, 
        {
            settings: ['site:news', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_en-US_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:es-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_es-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_es-US_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:es-ES'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_es-ES_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_es-ES_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:fr-FR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_fr-FR_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_fr-FR_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:it-IT'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_it-IT_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_it-IT_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:pt-BR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_pt-BR_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_pt-BR_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_zh-Hant-TW_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_zh-Hant-TW_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:news', 'lang:zh-Hant-HK'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_news_zh-Hant-HK_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_news_zh-Hant-HK_s_f_pw_351x40_news_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:news', 'partner:rogers', 'lang:en-CA'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/rogers_news_en-CA_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/rogers_news_en-CA_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:news', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:news', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:news', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_news_en-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_news_en-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:news', 'partner:att', 'lang:es-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_news_es-US_s_f_pw_351x40_news.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_news_es-US_s_f_pw_351x40_news_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
