// ycb/uh/tdv2-applet-uh/configs/services/theme/site/politics.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:politics', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_politics_en-US_s_f_pw_351x40_politics.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_politics_en-US_s_f_pw_351x40_politics_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:politics', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_politics_en-US_s_f_pw_351x40_politics.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_politics_en-US_s_f_pw_351x40_politics_2x.png',
                            partner: {
                                content_2: 'W(85px)'
                            },
                            yahoo: {
                                content_2: 'Mstart(85px)!'
                            }
                        }
                    }
                }
            }
        },
        {
            settings: ['site:politics', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_politics_en-US_s_f_pw_351x40_politics.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_politics_en-US_s_f_pw_351x40_politics_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:politics', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_politics_en-US_s_f_pw_351x40_politics.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_politics_en-US_s_f_pw_351x40_politics_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
