// ycb/uh/tdv2-applet-uh/configs/services/theme/site/sports.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:sports', 'lang:ar-JO'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_ar-JO_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_ar-JO_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_en-AE_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_en-AE_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'lang:en-AU'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_en-AU_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_en-AU_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_en-US_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_en-US_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'lang:zh-Hant-HK'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_zh-Hant-HK_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_zh-Hant-HK_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_sports_zh-Hant-TW_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_sports_zh-Hant-TW_s_f_pw_351x40_sports_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:sports', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_sports_en-US_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_sports_en-US_s_f_pw_351x40_sports_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_sports_en-US_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_sports_en-US_s_f_pw_351x40_sports_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:sports', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_sports_en-US_s_f_pw_351x40_sports.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_sports_en-US_s_f_pw_351x40_sports_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
