// ycb/uh/tdv2-applet-uh/configs/services/theme/site/style.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:style', 'lang:de-DE'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_de-DE_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_de-DE_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-AE_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-AE_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-AU'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-AU_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-AU_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-CA'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-CA_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-CA_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-GB'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-GB_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-GB_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-IN'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-IN_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-IN_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-SG'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-SG_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-SG_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_en-US_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_en-US_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:es-ES'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_es-ES_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_es-ES_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:es-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_es-US_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_es-US_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:fr-FR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_fr-FR_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_fr-FR_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:it-IT'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_it-IT_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_it-IT_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:pt-BR'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_pt-BR_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_pt-BR_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        {
            settings: ['site:style', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_style_zh-Hant-TW_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_style_zh-Hant-TW_s_f_pw_351x40_style_2x.png'
                        },
                        template: 'country_property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:style', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_style_en-US_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_style_en-US_s_f_pw_351x40_style_2x.png',
                            partner: {
                                content_2: 'W(85px)'
                            },
                            yahoo: {
                                content_2: 'Mstart(85px)!'
                            }
                        }
                    }
                }
            }
        },
        {
            settings: ['site:style', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_style_en-US_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_style_en-US_s_f_pw_351x40_style_2x.png'
                        }
                    }
                }
            }
        },
        // Remark: LOGO is not ready.
        // {
        //     settings: ['site:style', 'partner:rogers', 'lang:fr-CA'],
        //     uh: {
        //         logo: {
        //             style: {
        //                 cobrand: {
        //                     image_1x: 'https://s.yimg.com/rz/d/rogers_style_fr-CA_s_f_pw_351x40_style.png',
        //                     image_2x: 'https://s.yimg.com/rz/d/rogers_style_fr-CA_s_f_pw_351x40_style_2x.png'
        //                 }
        //             }
        //         }
        //     }
        // },
        {
            settings: ['site:style', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_style_en-US_s_f_pw_351x40_style.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_style_en-US_s_f_pw_351x40_style_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
