// ycb/uh/tdv2-applet-uh/configs/services/theme/site/tech.js
function ycb() {
    return [
        // --------------------------------------------------
        // LANG
        // --------------------------------------------------
        {
            settings: ['site:tech', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        property: {
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_tech_en-US_s_f_pw_351x40_tech.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_tech_en-US_s_f_pw_351x40_tech_2x.png'
                        },
                        template: 'property'
                    }
                }
            }
        },
        // --------------------------------------------------
        // PARTNER
        // --------------------------------------------------
        {
            settings: ['site:tech', 'partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/att_tech_en-US_s_f_pw_351x40_tech.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_tech_en-US_s_f_pw_351x40_tech_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:tech', 'partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/frontier_tech_en-US_s_f_pw_351x40_tech.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_tech_en-US_s_f_pw_351x40_tech_2x.png'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:tech', 'partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            image_1x: 'https://s.yimg.com/rz/d/verizon_tech_en-US_s_f_pw_351x40_tech.png',
                            image_2x: 'https://s.yimg.com/rz/d/verizon_tech_en-US_s_f_pw_351x40_tech_2x.png'
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
