// ycb/uh/tdv2-applet-uh/stores/MailStore.js
/**
 * @class MailStore
 */

'use strict';

var createStore = require('touchdown-v2/addons/createStore');
var MailStore = createStore({

    /**
     *  Name of the store
     *  @property storeName
     *  @default MailStore
     */
    storeName: 'MailStore',

    /**
     * Event handlers it listens to
     * @property handlers
     */
    statics: {
        handlers: {
            'NEW_MAIL_COUNT': '_receiveNewMailCount',
            'OPEN_MAILPREVIEW_SUCCESS': '_openMailPreviewSuccess',
            'OPEN_MAILPREVIEW_FAILURE': '_openMailPreviewFailure',
            'OPEN_MAILPREVIEW_NO_NEW': '_openMailPreviewNoNew'
        }
    },

    /**
     * Init store
     * @method initialize
     */
    initialize: function () {
        this.state = {
            mailcount: '',
            mailitems: null
        };
    },

    /**
     * Update the mailcount
     * @method _receiveNewMailCount
     */
    _receiveNewMailCount: function(count) {
        this.state.mailcount = count;
        this.emitChange();
    },

    /**
     * Accessor for the mailcount
     * @method getMailCount
     */
    getMailCount: function () {
        var count = 0;
        count = parseInt(this.state.mailcount, 10);
        return count;
    },

    /**
     * Accessor for mail preview
     * @method getMailPreview
     */
    getMailPreview: function () {
        return this.state.mailitems;
    },

    /**
     * Handler for mail preview items
     * @method _openMailPreviewSuccess
     */
    _openMailPreviewSuccess: function(payload) {
        this.state.mailitems = payload;
        this.emitChange();
    },

    /**
     * Handler for fetch failure
     * @method _openMailPreviewFailure
     */
    _openMailPreviewFailure: function () {
        this.state.mailitems = -1;
        this.emitChange();
    },

    /**
     * Handler for no new unread mail
     * @method _openMailPreviewNoNew
     */
    _openMailPreviewNoNew: function () {
        this.state.mailitems = [];
        this.emitChange();
    }
});

module.exports = MailStore;
