// ycb/uh/tdv2-applet-uh/stores/NavStore.js
/**
 * @class NavStore
 */

'use strict';

var createStore = require('fluxible-immutable-utils').createImmutableStore;
var debug = require('debug')('uh:stores:NavStore');
var Immutable = require('immutable');

var NavStore = createStore({

    storeName: 'NavStore',

    handlers: {
        NAV_FAILURE: '_handleNavFailure',
        NAV_SUCCESS: '_handleNavSuccess'
    },

    _handleNavFailure: function(error) {
        debug('_handleNavFailure', 'error', error);

        error = error || {};

        this.setState(this._state.withMutations(function updateState(state) {
            state.set('error', Immutable.fromJS(error));
        }));
    },

    _handleNavSuccess: function(payload) {
        debug('_handleNavSuccess', 'payload', payload);

        var data = payload && payload.data || {};
        this.setState(this._state.withMutations(function updateState(state) {
            state.set('data', Immutable.fromJS(data));
        }));
    },

    getError: function() {
        return this._state.getIn(['error']) || Immutable.fromJS({});
    },

    getItems: function() {
        return this._state.getIn(['data', 'items']) || Immutable.fromJS({});
    },

    getState: function() {
        return this._state;
    }
});

module.exports = NavStore;
