// ycb/uh/tdv2-applet-uh/configs/services/theme/theme/dark.js
function ycb() {
    return [
        {
            settings: ['theme:dark'],
            uh: {
                style: {
                    container: {
                        content: undefined,
                        right_list: 'End(20px) List(n) Pos(a) T(14px)',
                        wrapper_1: [
                            'Bdbs(s) Bdw(1px) Miw(980px) End(0)',
                            'Start(0) UH Z(10)'
                        ].join(' '),
                        wrapper_2: 'Bdc($c-divider) DarkTheme Py(14px)',
                        wrapper_3: 'T(0)',
                        wrapper_4: 'T(22px)',
                        wrapper_5: 'Pos(f)',
                        wrapper_6: 'Pos(r)',
                        wrapper_7: 'Bgc($bg-header)',
                        wrapper_8: 'Trs(shadowTransition) UhHideSubnav_Bxsh(shadowOn) Bxsh(shadowOff)'
                    }
                },
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: undefined,
                                content_2: undefined
                            },
                            wrapper_1: 'Bgr(nr) D(b) DarkTheme_Bgp(partnerLogoPos)! Fl(start) Mstart(10px)',
                            wrapper_2: undefined,
                            yahoo: {
                                content_1: undefined,
                                content_2: undefined
                            }
                        },
                        container: {
                            inline: {
                                backgroundColor: 'transparent'
                            },
                            wrapper_1: 'Fl(start) Fz(0) M(0) P(0)',
                            wrapper_2: 'Miw(190px) UHMR1D_Py(0px)'
                        },
                        country_property: {
                            content_1: 'Bgp(darkThemePos) Bgr(nr) Cur(p) D(b) H(35px)',
                            content_2: 'Bgz(702px) Mx(a)! W(155px)'
                        },
                        property: {
                            content_1: 'Bgp(darkThemePos) Bgr(nr) Cur(p) D(b) H(35px)',
                            content_2: 'Bgz(702px) Mx(a)! W(92px)',
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        template: 'yahoo',
                        yahoo: {
                            content_1: 'Bgpx(0) Bgr(nr) Cur(p) D(b) DarkTheme_Bgp(end_t) H(35px) Mstart(20px)',
                            content_2: 'Bgz(284px) Mx(a)! W(140px)',
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage_2x.png'
                        }
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_1: 'Bdrs(4px) D(ib) Px(14px) Py(5px) Td(n):h',
                            content_2: 'Bgc(darkthemeColor) Bgc(#fff):h',
                            icon_1: '#fff',
                            icon_2: '#000',
                            text_1: 'Fz(s) Lh(24px) Mstart(4px) Td(n) C(#fff)',
                            text_2: 'Fz(s) Lh(24px) Mstart(4px) Td(n) C(#000)',
                            wrapper: 'D(ib) Mstart(4px) Pb(4px) Va(t)'
                        },
                        mailcount: {
                            wrapper_1: 'C(#fff) Bgc(#f0162f) Fz(xs) Va(m) D(ib) Pos(a) Ta(c) Bdrs(4px) Mstart(3px) Mt(-11px) Py(3px) W(20px)'
                        }
                    }
                },
                profile: {
                    style: {
                        account_info: {
                            alias: 'C(textColor) Ell Maw(160px)',
                            content: 'C(#0078ff) C(#0078ff):h C(#0078ff):v D(ib) Fz(13px) My(2px)',
                            name: 'C(textColor) Ell Fw(b) Fz(13px) Maw(160px)',
                            wrapper: 'P(16px)'
                        },
                        avatar: {
                            content: 'Bdrs(100px) Bgz(cv) D(ib) H(34px) W(34px)',
                            color: '#fff',
                            name: 'D(n) Lh(2.6) Mstart(4px)',
                            wrapper: 'C(#fff) D(ib) Ell H(34px) Maw(120px)'
                        },
                        container: {
                            wrapper: 'Fl(start) H(34px) Mx(8px)'
                        },
                        panel: {
                            wrapper: [
                                'Bdrs(4px) Bgc(menuBgc) Bxsh(customShadowFlyoutMenu) D(n) End(0) Pos(a)',
                                'T(40px) Ta(start) uh-menu-open_D(b) W(280px)'
                            ].join(' ')
                        },
                        button: {
                            content_1: [
                                'Bdrs(4px) Bds(s) DarkTheme_Bds(n) Bdw(1px) C(#fff):h D(ib) Ell',
                                'Fil(n) Maw(120px) Px(26px) Py(8px) DarkTheme_Py(9px) Td(n):h'
                            ].join(' '),
                            content_2: [
                                'DarkTheme_Bdc(darkthemeColor) DarkTheme_Bdc(#fff):h',
                                'DarkTheme_Bgc(darkthemeColor) DarkTheme_Bgc(#fff):h DarkTheme_C(#000):h C(#fff)'
                            ].join(' '),
                            wrapper: 'Fl(start) Mx(4px)'
                        },
                        signed_out: {
                            content: 'C(textColor)',
                            wrapper: 'Bdtc(#e1e1e5) Bdts(s) Bdtw(1px) Mx(16px) Py(16px) Ta(c)'
                        }
                    }
                },
                search: {
                    style: {
                        cancel_button: {
                            content: 'D(n)'
                        },
                        container: {
                            content: 'M(0) P(0) Whs(nw)',
                            wrapper: 'uh-search D(ib) uh-max_Py(50px)'
                        },
                        input: {
                            wrapper_1: 'D(ib) Fl(start) Mend(6px)',
                            wrapper_2: 'Maw(486px) Miw(412px) W(42vw) ie-8_W(412px) ie-7_W(412px) ie-7_Pend(15px)'
                        },
                        search_button: {
                            content_1: [
                                'Bdrs(4px) Bdtw(0) Bdw(1px) Bgr(rx) Bxsh(customShadowSearchButton)',
                                'Bxz(cb) C(#fff) Ff(ss)! Fz(15px) Lh(32px)! M(0)! Miw(92px) Px(14px) Py(0)',
                                'Ta(c) Td(n) Va(t) Zoom'
                            ].join(' '),
                            content_2: 'Bg(#3775dd)',
                            inline: {
                                filter: 'chroma(color=#000000)'
                            }
                        }
                    }
                },
                topbar: {
                    style: {
                        nav_home: {
                            content: 'Bgc(#fff) C(#eee) List(n) My(.55em) Pos(r) Px(10px)!',
                            inline: {
                                boxShadow: '0 4px 7px rgba(0,0,0,.2)',
                                border: '0px 1px 1px 1px',
                                borderColor: '1px solid #d9d9d9'
                            },
                            link: {
                                content: 'C(#1d1da3)! D(b) Py(3px) Td(n) Td(u):h'
                            },
                            wrapper: 'Bd(1px solid #d9d9d9) Bgc(#fff) D(b) Pos(a) Start(0) T(100%)'
                        },
                        nav_main: {
                            content: 'H(22px) Lh(1.7) M(0) NavLinks P(0) Pos(r) Whs(nw) Z(11)',
                            link: {
                                content: 'C(#fff) Td(n) Td(u):h',
                                wrapper: 'D(ib) Lh(1.7) Mend(18px) Pstart(14px) Va(t) Zoom'
                            },
                            link_home: {
                                content_1: 'C(#fff) Td(n) Pos(r) Z(1) rapidnofollow',
                                content_2: 'C(#1d1da3)!',
                                icon: {
                                    down_arrow: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginLeft: '5px',
                                            verticalAlign: 'middle'
                                        }
                                    },
                                    home: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginRight: '6px',
                                            marginTop: '1px',
                                            verticalAlign: 'top'
                                        }
                                    }
                                },
                                text: 'Fw(400) Mstart(-1px) Td(u):h',
                                wrapper_1: 'D(ib) Lh(1.7) Mend(18px) Pstart(10px) Va(t) Zoom',
                                wrapper_2: 'Bgc(#fff)!'
                            },
                            link_more: {
                                content_1: 'C(#fff) Pos(r) Td(n) Z(1) yucs-leavable rapidnofollow',
                                content_2: 'C(#1d1da3)!',
                                icon: {
                                    down_arrow: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginLeft: '5px',
                                            verticalAlign: 'middle'
                                        }
                                    }
                                },
                                text: 'Td(u):h',
                                wrapper_1: 'D(ib) Pend(6px) Pos(r) Pstart(10px) Va(t) Z(4) Zoom',
                                wrapper_2: 'Bgc(#fff)!'
                            },
                            wrapper_1: 'C(#fff) Fz(13px)',
                            wrapper_2: 'Bgc(#2d1152)'
                        },
                        nav_more: {
                            content: 'Bgc(#fff) C(#eee) Pos(r) Px(10px)! List(n) My(.55em)',
                            inline: {
                                boxShadow: '0 4px 7px rgba(0,0,0,.2)',
                                border: '0px 1px 1px 1px',
                                borderColor: '1px solid #d9d9d9'
                            },
                            link: {
                                content: 'C(#1d1da3)! D(b) Py(3px) Td(n) Td(u):h'
                            },
                            wrapper: 'Bgc(#fff) Bd(1px solid #d9d9d9) D(b) Pos(a) Start(0) T(100%)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:ar-JO'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_1: 'Bgpx(-552px) Bgr(nr) Cur(p) D(b) H(35px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(180px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:style', 'theme:dark', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(140px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['site:movies', 'theme:dark', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(158px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:en-AU'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(115px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:zh-Hant-HK'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(130px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(702px) Mx(a)! W(130px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:zh-Hant-TW', 'site:news'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_1: 'Bgp(darkThemePos) Bgr(nr) Cur(p) D(b) H(37px)',
                            content_2: 'Bgz(702px) Mx(a)! W(125px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'lang:zh-Hant-TW', 'site:style'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_1: 'Bgp(darkThemePos) Bgr(nr) Cur(p) D(b) H(37px)',
                            content_2: 'Bgz(702px) Mx(a)! W(155px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'device:smartphone'],
            uh: {
                search: {
                    style: {
                        cancel_button: {
                            content: 'C(#007aff) End(0) Fw(500) Fz(14px) Pos(a) Px(10px) Py(5px) uh-search-open_V(v) V(h) Va(m)'
                        },
                        container: {
                            wrapper: [
                                'BdB Bdc(#b5b5b5) Bgc(#fff) D(b) Py(8px)',
                                'Start(0) Mah(33px) W(100%) Z(4)'
                            ].join(' ')
                        },
                        input: {
                            content: 'Pstart(20px)',
                            wrapper_1: 'Bxz(bb) D(ib) Pend(60px) Pstart(10px)',
                            wrapper_2: 'W(100%)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    }
                },
                mail: {
                    style: {
                        mailcount: {
                           wrapper_1: [ 'Bdrs(50%)! Bds(s) Bdw(3px) Fw(b) Lh(1.7) H(18px) Mstart(-13px) Mt(3px) Pos(a) Py(0) W(18px)',
                                      'C(#fff) Bgc(#f0162f) Fz(xs) Va(m) D(ib) Ta(c)'
                           ].join(' ')
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'device:smartphone', 'lang:de-DE' ],
            uh: {
                search: {
                    style: {
                        input: {
                            wrapper_1: 'Bxz(bb) D(ib) Pend(83px) Pstart(10px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:dark', 'device:tablet'],
            uh: {
                mail: {
                    style: {
                        mailcount: {
                           wrapper_1: [ 'Bdrs(50%)! Bds(s) Bdw(3px) Fw(b) Lh(1.7) H(18px) Mstart(12px) T(3px) Pos(a) Py(0) W(18px)',
                                      'C(#fff) Bgc(#f0162f) Fz(xs) Va(m) D(ib) Ta(c)'
                           ].join(' ')
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
