// ycb/uh/tdv2-applet-uh/configs/services/theme/theme/mega.js
function ycb() {
    return [
        {
            settings: ['theme:mega'],
            uh: {
                style: {
                    container: {
                        content: 'Maw(1256px) Miw(1090px) Mx(a)',
                        right_list: 'Fl(end) List(n) Mend(17px) nonsticky_T(3px) Pos(r) T(3px)',
                        wrapper_1: [
                            'Bxsh(shadowOff) Miw(980px) End(0)',
                            'Start(0) Trs(shadowTransition) UH UhHideSubnav_Bxsh(shadowOn) Z(10)'
                        ].join(' '),
                        wrapper_2: [
                            'Bdt(customBorderTop) MegaTheme Pb(22px) Pt(24px)',
                            'Trstf(eo) Trsde(0s) Trsdu(0.13s) Trsp(a)',
                            'nonsticky_Pb(2px) nonsticky_Pt(13px) nonsticky_Bxsh(headerShadow)'
                        ].join(' '),
                        wrapper_3: 'T(0)',
                        wrapper_4: 'T(22px)',
                        wrapper_5: 'Pos(f)',
                        wrapper_6: 'Pos(r)',
                        wrapper_7: 'Bgc(#fff)'
                    }
                },
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: undefined,
                                content_2: undefined
                            },
                            wrapper_1: 'Bgr(nr) D(b) Fl(start) Mstart(10px)',
                            wrapper_2: undefined,
                            yahoo: {
                                content_1: undefined,
                                content_2: undefined
                            }
                        },
                        container: {
                            inline: {
                                backgroundColor: 'transparent'
                            },
                            wrapper_1: 'Fl(start) Fz(0) M(0) P(0)',
                            wrapper_2: 'Miw(210px) UHMR1D_Py(0px)'
                        },
                        country_property: {
                            content_1: 'Bgpx(0) Bgr(nr) Cur(p) D(b) H(48px)',
                            content_2: 'Bgz(880px) Mx(a)! W(210px)'
                        },
                        property: {
                            content_1: 'Bgpx(0) Bgr(nr) Cur(p) D(b) H(40px)',
                            content_2: 'Bgz(880px) Mx(a)! W(120px)',
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        template: 'yahoo',
                        yahoo: {
                            content_1: [
                                'Bgpx(4px) Bgpy(-3px) Bgpy(7px)!--sm1024 Bgr(nr) Cur(p) D(ib) Fl(start) H(48px)',
                                'nonsticky_Bgpy(7px)--sm1024 W(160px)'
                            ].join(' '),
                            content_2: ['Bgz(157px)  Bgz(90px)!--sm1024 W(160px) Mend(29px) Mend(31px)!--sm1024',
                                       'Mstart(12px) Mstart(10px)!--sm1024'
                            ].join(' '),
                            image_1x: 'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_p_bestfit_frontpage.png',
                            image_2x: 'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_p_bestfit_frontpage_2x.png'
                        }
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_1: 'Pos(r) D(ib) Ta(s) Td(n):h',
                            content_2: '',
                            icon_1: '#400090',
                            icon_2: '#400090',
                            inline: {
                                WebkitFontSmoothing: 'auto'
                            },
                            text_1: 'Lh(userNavTextLh) D(ib) C(mailBtn) Fz(14px) Fw(b) Va(t) Mstart(13px)',
                            text_2: 'Lh(userNavTextLh) D(ib) C(mailBtn) Fz(14px) Fw(b) Va(t) Mstart(13px)',
                            wrapper: 'D(ib) Mstart(14px) Mt(-1px) ua-ie8_Pb(10px) ua-ie9_Pb(10px)'
                        },
                        mailcount: {
                            wrapper_1: 'Bgc(mailBadge) Bdrs(11px) C(#fff) Start(19px) Fz(xs) Fw(b) Pos(a) Py(4px) Ta(c) T(-7px) W(22px)'
                        }
                    }
                },
                profile: {
                    style: {
                        account_info: {
                            alias: 'C(textColor) Ell Maw(160px)',
                            content: 'C(#0078ff) C(#0078ff):h C(#0078ff):v D(ib) Fz(13px) My(2px)',
                            name: 'C(textColor) Ell Fw(b) Fz(13px) Maw(160px)',
                            wrapper: 'P(16px)'
                        },
                        avatar: {
                            content: [ 'Lh(userNavTextLh) Bdrs(100px) Bgz(cv) C(signInBtn)',
                                       'Cur(p) D(ib) Fz(34px) H(34px) W(34px) ua-ie7_D(n)'
                            ].join(' '),
                            color: '#400090',
                            name: [ 'D(ib) Maw(100px) Ov(h) Tov(e) Lh(userNavTextLh) C(mailBtn)',
                                    'Pstart(8px) Fz(14px) Fw(b) Va(t) ua-ie7_D(n)'
                            ].join(' '),
                            wrapper: 'P(0) Bd(0) Pos(r) ua-ie8_Pb(10px) ua-ie9_Pb(10px)'
                        },
                        container: {
                            wrapper: 'Fl(start) H(34px) Mx(8px)'
                        },
                        panel: {
                            wrapper: [
                                'Bdrs(4px) Bgc(menuBgc) Bxsh(customShadowFlyoutMenu) D(n) End(0) Pos(a)',
                                'T(40px) Ta(start) uh-menu-open_D(b) W(280px)'
                            ].join(' ')
                        },
                        button: {
                            content_1: [
                                'Bdrs(5px) Bds(s) Bdw(2px) C(#fff):h D(ib) Ell Fz(14px)',
                                'Fw(b) Py(2px) Mt(5px) Ta(c) Td(n):h Miw(78px) H(18px)'
                            ].join(' '),
                            content_2: 'Bdc(signInBtn) Bgc(signInBtn):h C(signInBtn)',
                            inline: {
                                WebkitFontSmoothing: 'auto'
                            },
                            wrapper: 'Fl(start) Mx(4px) Mend(9px)'
                        },
                        signed_out: {
                            content: 'C(textColor)',
                            wrapper: 'Bdtc(#e1e1e5) Bdts(s) Bdtw(1px) Mx(16px) Py(16px) Ta(c)'
                        }
                    }
                },
                search: {
                    style: {
                        cancel_button: {
                            content: 'D(n)'
                        },
                        container: {
                            content: 'M(0) P(0)',
                            wrapper: [
                                'Mt(-2px) D(b) Fl(start) Mstart(-4px) Mstart(-90px)--sm1024',
                                'Trstf(eo) Trsde(0s) Trsdu(0.14s) Trsp(a) uh-max_Py(50px) W(54%)'
                            ].join(' ')
                        },
                        input: {
                            content: ['Bdc(searchBdr) glow_Bdc(searchBtnGlow) Bdc(searchBdrFoc):f',
                                      'Bdrs(2px)! ua-ie7_H(36px) ua-ie7_Lh(36px) ua-ie8_Lh(36px) ua-ie7_W(92%)',
                                      'H(40px) nonsticky_H(40px) Px(10px) glowing_Trs(searchMegaTransition)'].join (' '),
                            wrapper_1: 'D(ib) Fl(start) Mend(5px)',
                            wrapper_2: 'W(66.55%) W(401px)--sm1024'
                        },
                        search_button: {
                            content_1: [
                                'Bdrs(3px) Bdw(0) glowing_Trs(searchMegaTransition) Bxsh(customShadowMegaSearchButton)',
                                'glow_Bgc(searchBtnGlow) Bxsh(searchBtnShadow) glow_Bxsh(searchBtnShadowGlow)',
                                'C(#fff) Fz(16px) H(40px) Lh(2px) Fw(500px) Whs(nw) P(0)! M(0)!',
                                'nonsticky_H(40px) Py(0) Ta(c) Td(n) Va(t) W(140px) Zoom'
                            ].join(' '),
                            content_2: 'Bgc(searchBtn) Bgc(searchBtnHvr):h',
                            inline: {
                                filter: 'chroma(color=#000000)',
                                WebkitFontSmoothing: 'antialiased'
                            }
                        }
                    }
                },
                topbar: {
                    style: {
                        nav_home: {
                            content: 'Bgc(#fff) C(#eee) List(n) My(.55em) Pos(r) Px(10px)!',
                            inline: {
                                boxShadow: '0 4px 7px rgba(0,0,0,.2)',
                                border: '0px 1px 1px 1px',
                                borderColor: '1px solid #d9d9d9'
                            },
                            link: {
                                content: 'C(#1d1da3)! D(b) Py(3px) Td(n) Td(u):h'
                            },
                            wrapper: 'Bd(1px solid #d9d9d9) Bgc(#fff) D(b) Pos(a) Start(0) T(100%)'
                        },
                        nav_main: {
                            content: 'H(22px) Lh(1.7) M(0) NavLinks P(0) Pos(r) Whs(nw) Z(11)',
                            link: {
                                content: 'C(#fff) Td(n) Td(u):h',
                                wrapper: 'D(ib) Lh(1.7) Mend(18px) Pstart(14px) Va(t) Zoom'
                            },
                            link_home: {
                                content_1: 'C(#fff) Td(n) Pos(r) Z(1) rapidnofollow readerOpen_V(h)',
                                content_2: 'C(#1d1da3)!',
                                icon: {
                                    down_arrow: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginLeft: '5px',
                                            verticalAlign: 'middle'
                                        }
                                    },
                                    home: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginRight: '6px',
                                            marginTop: '1px',
                                            verticalAlign: 'top'
                                        }
                                    }
                                },
                                text: 'Fw(400) Mstart(-1px) Td(u):h',
                                wrapper_1: 'D(ib) Lh(1.7) Mend(18px) Pstart(10px) Va(t) Zoom',
                                wrapper_2: 'Bgc(#fff)!'
                            },
                            link_more: {
                                content_1: 'C(#fff) Pos(r) Td(n) Z(1) yucs-leavable rapidnofollow readerOpen_V(h)',
                                content_2: 'C(#1d1da3)!',
                                icon: {
                                    down_arrow: {
                                        color_1: '#fff',
                                        color_2: '#1d1da3',
                                        inline: {
                                            cursor: 'pointer',
                                            marginLeft: '5px',
                                            verticalAlign: 'middle'
                                        }
                                    }
                                },
                                text: 'Td(u):h',
                                wrapper_1: 'D(ib) Pend(6px) Pos(r) Pstart(10px) Va(t) Z(4) Zoom',
                                wrapper_2: 'Bgc(#fff)!'
                            },
                            wrapper_1: 'C(#fff) Fz(13px)',
                            wrapper_2: 'Bgc(#2d1152)'
                        },
                        nav_more: {
                            content: 'Bgc(#fff) C(#eee) Pos(r) Px(10px)! List(n) My(.55em)',
                            inline: {
                                boxShadow: '0 4px 7px rgba(0,0,0,.2)',
                                border: '0px 1px 1px 1px',
                                borderColor: '1px solid #d9d9d9'
                            },
                            link: {
                                content: 'C(#1d1da3)! D(b) Py(3px) Td(n) Td(u):h'
                            },
                            wrapper: 'Bgc(#fff) Bd(1px solid #d9d9d9) D(b) Pos(a) Start(0) T(100%)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:ar-JO'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_1: 'Bgpx(-238px) Bgr(nr) Cur(p) D(b) H(40px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:en-AE'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(880px) Mx(a)! W(225px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:en-AU'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(880px) Mx(a)! W(145px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:zh-Hant-HK'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(880px) Mx(a)! W(162px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(880px) Mx(a)! W(158px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'lang:zh-Hant-TW', 'site:style'],
            uh: {
                logo: {
                    style: {
                        country_property: {
                            content_2: 'Bgz(880px) Mx(a)! W(210px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'device:smartphone'],
            uh: {
                search: {
                    style: {
                        cancel_button: {
                            content: 'C(#007aff) End(0) Fw(500) Fz(14px) Pos(a) Px(10px) Py(5px) uh-search-open_V(v) V(h) Va(m)'
                        },
                        container: {
                            wrapper: [
                                'BdB Bdc(#b5b5b5) Bgc(#fff) D(b) Py(8px)',
                                'Start(0) Mah(33px) W(100%) Z(4)'
                            ].join(' ')
                        },
                        input: {
                            content: 'Pstart(20px)',
                            wrapper_1: 'Bxz(bb) D(ib) Pend(60px) Pstart(10px)',
                            wrapper_2: 'W(100%)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    }
                },
                mail: {
                    style: {
                        mailcount: {
                           wrapper_1: [ 'Bdrs(50%)! Bds(s) Bdw(3px) Fw(b) Lh(1.7) H(18px) Mstart(-13px) Mt(3px) Pos(a) Py(0) W(18px)',
                                      'C(#fff) Bgc(#f0162f) Fz(xs) Va(m) D(ib) Ta(c)'
                           ].join(' ')
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'device:smartphone', 'lang:de-DE' ],
            uh: {
                search: {
                    style: {
                        input: {
                            wrapper_1: 'Bxz(bb) D(ib) Pend(83px) Pstart(10px)'
                        }
                    }
                }
            }
        },
        {
            settings: ['theme:mega', 'device:tablet'],
            uh: {
                mail: {
                    style: {
                        mailcount: {
                           wrapper_1: [ 'Bdrs(50%)! Bds(s) Bdw(3px) Fw(b) Lh(1.7) H(18px) Mstart(12px) T(3px) Pos(a) Py(0) W(18px)',
                                      'C(#fff) Bgc(#f0162f) Fz(xs) Va(m) D(ib) Pos(a) Ta(c)'
                           ].join(' ')
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
