// ycb/uh/tdv2-applet-uh/actions/loadHeader.js
'use strict';

var debug = require('debug')('uh:actions:loadHeader');
var executeMultiple = require('fluxible-action-utils/async/executeMultiple');

/**
 * Initialize action for a header
 * @param {Object} context request context
 * @param {Object} payload metadata passed in by component
 * @param {Function} done callback function
 */
module.exports = function initialize(context, payload, done) {
    debug('Loading the Header');
    var routeStore = context.getStore('RouteStore');
    var data = {};

    data.payload = payload;
    context.dispatch('UH_NAVIGATE_INIT_ON_START', routeStore.getCurrentRoute());

    executeMultiple(
        context,
        {
            loadProfile: {
                action: require('../actions/loadProfile'),
                params: payload
            },
            loadNav: {
                action: require('../actions/loadNav'),
                params: payload
            },
            loadTheme: {
                action: require('../actions/loadTheme'),
                params: payload
            }
        },
        function (err, data) {
            if (err) {
                if (err.loadProfile) {
                    // TODO: Dispatch LOAD_PROFILE_FAILURE
                    // context.dispatch('LOAD_PROFILE_FAILURE', err.loadProfile);
                }
                if (err.loadNav) {
                    // TODO: Dispatch LOAD_NAV_FAILURE
                    // context.dispatch('LOAD_NAV_FAILURE', err.loadNav);
                }
                if (err.loadTheme) {
                    // TODO: Dispatch LOAD_THEME_FAILURE
                    // context.dispatch('LOAD_THEME_FAILURE', err.loadTheme);
                }
                return done && done(err, null);
            }

            return done && done(null, data);
        }
    );
};
