// ycb/uh/tdv2-applet-uh/configs/components/partner/index.js
function ycb() {
    return [
        {
            settings: ['partner:att', 'lang:en-US'],
            uh: {
                theme:{
                    topbarBackground: 'Bgc(#067ab4)',
                    topbarIcon: '#067ab4',
                    profileBackground: 'Bgc(#067ab4):h C(#067ab4) Bdc(#067ab4)',
                    profileIcon: '#067ab4',
                    mailBackground: 'Bgc(#067ab4)',
                    mailNotificationBackground: 'Bgc(#f0162f)'
                },
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage:'https://s.yimg.com/rz/d/att_en-US_f_pw_351x40_2x.png',
                    defaultNonRetinaImage:'https://s.yimg.com/rz/d/att_en-US_f_pw_351x40.png',
                    partnerLink: 'http://att.net/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(95px)',
                    logoRightWidth: 'W(75px)',
                    logoMarginLeft: 'Mstart(95px)!',
                    height: 'H(31px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(partnerLogoPos)!'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'sbc'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        'signed_in': {
                            query: {
                                '.partner': 'sbc'
                            }
                        },
                        'signed_out': {
                            query: {
                                '.done': 'https://att.yahoo.com/',
                                '.partner': 'sbc'
                            }
                        },
                        'account_info': {
                            query: {
                                '.partner': 'sbc'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'att',
                        hsimp: 'yhs-att_001',
                        type: 'sbc_dsl'
                    }
                }
            }
        },
        {
            settings: ['partner:att', 'lang:es-US'],
            uh: {
                theme:{
                    topbarBackground: 'Bgc(#067ab4)',
                    topbarIcon: '#067ab4',
                    profileBackground: 'Bgc(#067ab4):h C(#067ab4) Bdc(#067ab4)',
                    profileIcon: '#067ab4',
                    mailBackground: 'Bgc(#067ab4)',
                    mailNotificationBackground: 'Bgc(#f0162f)'
                },
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage: 'https://s.yimg.com/rz/d/att_es-US_f_pw_351x40_2x.png',
                    defaultNonRetinaImage: 'https://s.yimg.com/rz/d/att_es-US_f_pw_351x40.png',
                    link: 'http://envivo.att.com/',
                    partnerLink: 'http://envivo.att.com/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(95px)',
                    logoRightWidth: 'W(75px)',
                    logoMarginLeft: 'Mstart(95px)!',
                    height: 'H(28px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(attEspanolLogoPos)!'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'sbc'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        'signed_in': {
                            query: {
                                '.partner': 'sbc'
                            }
                        },
                        'signed_out': {
                            query: {
                                '.done': 'https://espanol.att.yahoo.com/',
                                '.partner': 'sbc'
                            }
                        },
                        'account_info': {
                            query: {
                                '.partner': 'sbc'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://espanol.search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'att',
                        hsimp: 'yhs-att_001',
                        type: 'sbc_dsl'
                    }
                },
                topbar: {
                    data: {
                        more: 11
                    }
                }
            }
        },
        {
            settings: ['partner:verizon', 'lang:en-US'],
            uh: {
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage: 'https://s.yimg.com/rz/d/verizon_en-US_f_pw_351x40_2x.png',
                    defaultNonRetinaImage: 'https://s.yimg.com/rz/d/verizon_en-US_f_pw_351x40.png',
                    partnerLink: 'https://verizon.yahoo.com/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(67px)',
                    logoRightWidth: 'W(97px)',
                    logoMarginLeft: 'Mstart(67px)!',
                    height: 'H(35px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(partnerLogoPos)!'
                    // paddingY: 'Py(0px)'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'vz-acs'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        signed_out: {
                            query: {
                                '.done': 'https://verizon.yahoo.com/',
                                '.partner': 'vz-acs'
                            }
                        },
                        account_info: {
                            query: {
                                '.partner': 'vz-acs'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'vz',
                        hsimp: 'yhs-vz',
                        type: 'verizon_access'
                    }
                }
            }
        },
        {
            settings: ['partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_351x40_2x.png',
                    defaultNonRetinaImage: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_351x40.png',
                    partnerLink: 'https://frontier.yahoo.com/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(85px)',
                    logoRightWidth: 'W(85px)',
                    logoMarginLeft: 'Mstart(85px)!',
                    height: 'H(31px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(partnerLogoPos)!'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'ftr'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        'signed_out': {
                            query: {
                                '.done': 'https://frontier.yahoo.com/',
                                '.partner': 'ftr'
                            }
                        },
                        'account_info': {
                            query: {
                                '.partner': 'ftr'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'frontier',
                        hsimp: 'yhs-frontier1',
                        type: 'frontier-cobrand'
                    }
                }
            }
        },
        {
            settings: ['partner:rogers', 'lang:en-CA'],
            uh: {
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40_2x.png',
                    defaultNonRetinaImage:'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40.png',
                    link: 'https://ca.rogers.yahoo.com/',
                    partnerLink: 'https://ca.rogers.yahoo.com/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(90px)',
                    logoRightWidth: 'W(80px)',

                    logoMarginLeft: 'Mstart(90px)!',
                    height: 'H(28px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(partnerLogoPos)!'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'rogers-acs'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        'signed_out': {
                            query: {
                                '.done': 'https://ca.rogers.yahoo.com/',
                                '.partner': 'rogers-acs'
                            }
                        },
                        'account_info': {
                            query: {
                                '.partner': 'rogers-acs'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://ca.search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'rogers',
                        hsimp: 'yhs-rogers_001',
                        type: 'rogers_hispeed'
                    }
                },
                topbar: {
                    data: {
                        more: 9
                    }
                }
            }
        },
        {
            settings: ['partner:rogers', 'lang:fr-CA'],
            uh: {
                logo: {
                    // data
                    isCobrand: true,
                    defaultRetinaImage: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40_2x.png',
                    defaultNonRetinaImage:'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40.png',
                    link: 'https://fr-ca.rogers.yahoo.com/',
                    partnerLink: 'https://fr-ca.rogers.yahoo.com/',

                    // style
                    logoBgSize: 'Bgz(702px)',
                    logoLeftWidth: 'W(90px)',
                    logoRightWidth: 'W(80px)',
                    logoMarginLeft: 'Mstart(90px)!',
                    height: 'H(28px)',
                    width: 'W(170px)',
                    backgroundPosition: 'DarkTheme_Bgp(partnerLogoPos)!'
                },
                mail: {
                    urls: {
                        mail: {
                            query: {
                                '.partner': 'rogers-acs'
                            }
                        }
                    }
                },
                profile: {
                    urls: {
                        'signed_out': {
                            query: {
                                '.done': 'https://fr-ca.rogers.yahoo.com/',
                                '.partner': 'rogers-acs'
                            }
                        },
                        'account_info': {
                            query: {
                                '.partner': 'rogers-acs'
                            }
                        }
                    }
                },
                search: {
                    action: 'https://ca.search.yahoo.com/yhs/search',
                    queries: {
                        hspart: 'rogers',
                        hsimp: 'yhs-rogers_001',
                        type: 'rogers_hispeed'
                    }
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:att', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#000)',
                    mobileHeaderIcon: '#067ab4'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/att_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'http://att.net/',
                    text: 'AT&T',

                    // style
                    backgroundColor: 'transparent',
                    width: 'W(133px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-13px)!',
                    logoLeftWidth: 'W(74px)',
                    logoMarginRight: 'Mstart(60px)!',
                    logoRightWidth: 'W(60px)'
                },
                mail: {
                    url: 'https://mlogin.yahoo.com/w/login/?.src=fpctx'+
                    '&.intl=us&.partner=sbc&.lang=en-US'+
                    '&.done=https://m.yahoo.com/w/ygo-mail/?.tsrc=attmp',
                    signedInUrl: "https://m.yahoo.com/w/ygo-mail/home.bp?.intl=us&.lang=en&.tsrc=attmp"
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:att', 'lang:es-US'],
            uh: {
                theme:{
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#000)',
                    mobileHeaderIcon: '#067ab4'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/att_es-US_f_pw_bestfit.png',
                    partnerLink: 'http://envivo.att.com/',
                    text: 'AT&T',

                    // style
                    width: 'W(155px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-24px)!',
                    logoLeftWidth: 'W(80px)',
                    logoMarginRight: 'Mstart(56px)!',
                    logoRightWidth: 'W(62px)'
                },
                mail: {
                    url: 'https://mail.yahoo.com/?.partner=sbc&.intl=e1&.lang=es-US'
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:verizon', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/verizon_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://verizon.yahoo.com/',
                    text: 'Verizon',

                    // style
                    width: 'W(112px)',
                    height: 'H(23px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-1px)!',
                    logoLeftWidth: 'W(45px)',
                    logoMarginRight: 'Mstart(43px)!',
                    logoRightWidth: 'W(67px)'
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:frontier', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://frontier.yahoo.com/',
                    text: 'Frontier',

                    // style
                    width: 'W(138px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-15px)!',
                    logoLeftWidth: 'W(69px)',
                    logoMarginRight: 'Mstart(54px)!',
                    logoRightWidth: 'W(69px)'
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:rogers', 'lang:en-CA'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://ca.rogers.yahoo.com/',
                    text: 'Rogers',

                    // style
                    width: 'W(141px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-12px)!',
                    logoLeftWidth: 'W(70px)',
                    logoMarginRight: 'Mstart(59px)!',
                    logoRightWidth: 'W(68px)'
                }
            }
        },
        {
            settings: ['device:smartphone', 'partner:rogers', 'lang:fr-CA'],
            uh: {
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://fr-ca.rogers.yahoo.com/',
                    text: 'Rogers',

                    // style
                    width: 'W(141px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-12px)!',
                    logoLeftWidth: 'W(70px)',
                    logoMarginRight: 'Mstart(59px)!',
                    logoRightWidth: 'W(68px)'
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:att', 'lang:es-US'],
            uh: {
                theme:{
                    tabletHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#067ab4)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#067ab4'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/att_es-US_f_pw_bestfit.png',
                    partnerLink: 'http://envivo.att.com/',
                    text: 'AT&T',

                    // style
                    width: 'W(155px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-24px)!',
                    logoLeftWidth: 'W(80px)',
                    logoMarginRight: 'Mstart(56px)!',
                    logoRightWidth: 'W(62px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:att', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    tabletHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileHeaderText: 'C(#067ab4)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#067ab4'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/att_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'http://att.net/',
                    text: 'AT&T',

                    // style
                    backgroundColor: 'transparent',
                    width: 'W(133px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-13px)!',
                    logoLeftWidth: 'W(74px)',
                    logoMarginRight: 'Mstart(60px)!',
                    logoRightWidth: 'W(60px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:frontier', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://frontier.yahoo.com/',
                    text: 'Frontier',

                    // style
                    width: 'W(138px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-15px)!',
                    logoLeftWidth: 'W(69px)',
                    logoMarginRight: 'Mstart(54px)!',
                    logoRightWidth: 'W(69px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:verizon', 'lang:en-US'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/verizon_en-US_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://verizon.yahoo.com/',
                    text: 'Verizon',

                    // style
                    width: 'W(112px)',
                    height: 'H(23px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-1px)!',
                    logoLeftWidth: 'W(45px)',
                    logoMarginRight: 'Mstart(43px)!',
                    logoRightWidth: 'W(67px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:rogers', 'lang:en-CA'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://ca.rogers.yahoo.com/',
                    text: 'Rogers',

                    // style
                    width: 'W(141px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-12px)!',
                    logoLeftWidth: 'W(70px)',
                    logoMarginRight: 'Mstart(59px)!',
                    logoRightWidth: 'W(68px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        },
        {
            settings: ['device:tablet', 'partner:rogers', 'lang:fr-CA'],
            uh: {
                theme:{
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    mobileHeaderBackground: 'Bgc(#fff) Bdbc(#edebec) Bdbs(s) Bdw(1px)',
                    mobileLogoText: 'C(#000)',
                    mobileHeaderIcon: '#6248df'
                },
                logo: {
                    // data
                    image: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_bestfit.png',
                    link: false, // Set to false so that link is created dynamically in LogoMobile.jsx
                    partnerLink: 'https://fr-ca.rogers.yahoo.com/',
                    text: 'Rogers',

                    // style
                    width: 'W(141px)',
                    height: 'H(21px)',
                    backgroundPosition: '0',
                    logoMarginLeft: 'Mstart(-12px)!',
                    logoLeftWidth: 'W(70px)',
                    logoMarginRight: 'Mstart(59px)!',
                    logoRightWidth: 'W(68px)'
                },
                search: {
                    style: {
                        input: {
                            content: 'Bdrs(4px)',
                            wrapper_2: 'W(40vw)'
                        },
                        search_button: {
                            content_1: 'D(n)'
                        }
                    },
                    autofocus: false,
                    placeholderText: 'SEARCH_WEB',
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                }
            }
        }
    ];
}

module.exports = ycb();
