// ycb/uh/tdv2-applet-uh/configs/components/master.js
function ycb() {
    return [
        {
            settings: ['master'],
            i13n: {
                sec: 'uh'
            },
            uh: {
                // uh: style
                style: {
                    container: {
                        content: undefined,
                        right_list: undefined,
                        wrapper_1: undefined,
                        wrapper_2: undefined,
                        wrapper_3: undefined,
                        wrapper_4: undefined,
                        wrapper_5: undefined,
                        wrapper_6: undefined,
                        wrapper_7: undefined,
                        wrapper_8: undefined
                    }
                },
                backgroundColor: 'Bgc(#f9f9f9)',
                theme:{
                    topbarBackground: 'Bgc(#2d1152)',
                    topbarIcon: '#1d1da3',
                    profileBackground: 'Bgc(uhPurple):h C(uhPurple) Bdc(uhPurple)',
                    profileIcon: '#400090',
                    mailBackground: 'Bgc(uhPurple)',
                    mailNotificationBackground: 'Bgc(#f0162f)',
                    mailNotificationBorderColor: 'Bdc(#fff)',
                    followNotificationBorderColor: 'Bdc(#fff)',
                    // tablet theme
                    tabletHeaderBackground: undefined,
                    // mobile theme
                    mobileHeaderBackground: undefined,
                    mobileLogoText: undefined,
                    mobileHeaderText: undefined,
                    mobileHeaderIcon: undefined
                },

                // uh: components
                help: {
                    // help: style
                    style: {
                        container: {
                            wrapper: 'D(ib) Mstart(16px) Mt(3px)'
                        },
                        separator: {
                            content: 'Bgc(#e2e2e6) H(1px)'
                        },
                        icon: {
                            settings: {
                                color: '#400090',
                                content: 'D(n) Lh(2.6) Mstart(4px)',
                                name: 'CoreSettings',
                                size: 28,
                                wrapper: 'D(ib)'
                            }
                        },
                        link: {
                            content: 'Td(n) Td(u):h D(b) C(#000)',
                            wrapper: 'Py(8px) Px(10px)'
                        },
                        panel: {
                            content: 'Mx(-10px) Pos(r) My(0) P(0) C(#000)!',
                            wrapper: [
                                'Bdrs(4px) Bgc(menuBgc) Bxsh(customShadowFlyoutMenu) D(b) End(0) Pos(a)',
                                'Px(10px) T(40px) Ta(start)'
                            ].join(' ')
                        }
                    },
                    // help: data
                    data: {
                        items: {
                            account_info: {
                                // link: 'https://login.yahoo.com/account/personalinfo?.intl=us&.lang=en-US&.done=http://news.yahoo.com/&.src=yn',
                                order: 1,
                                title: 'Account Info',
                                url: {
                                    protocol: 'https',
                                    hostname: 'login.yahoo.com',
                                    pathname: '/account/personalinfo/',
                                    query: {
                                        '.done': undefined,
                                        '.intl': undefined,
                                        '.lang': undefined,
                                        '.partner': undefined,
                                        '.src': undefined
                                    }
                                }
                            },
                            help: {
                                // link: 'https://help.yahoo.com/l/us/yahoo/news_global/',
                                order: 2,
                                title: 'Help',
                                url: {
                                    protocol: 'https',
                                    hostname: 'help.yahoo.com',
                                    pathname: '/l/us/yahoo/news_global/'
                                }
                            },
                            separator: {
                                order: 3
                            },
                            suggestions: {
                                // link: 'http://feedback.yahoo.com/forums/204992',
                                order: 4,
                                title: 'Suggestions',
                                url: {
                                    protocol: 'https',
                                    hostname: 'yahoo.uservoice.com',
                                    pathname: '/forums/204992/'
                                }
                            }
                        }
                    }
                },
                logo: {
                    // logo: style
                    style: {
                        cobrand: {
                            partner: {
                                content_1: undefined,
                                content_2: undefined
                            },
                            image_1x: undefined,
                            image_2x: undefined,
                            wrapper_1: undefined,
                            wrapper_2: undefined,
                            yahoo: {
                                content_1: undefined,
                                content_2: undefined
                            }
                        },
                        container: {
                            inline: {
                                backgroundColor: undefined
                            },
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        country_property: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        property: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        // template: [cobrand | country_property | property | yahoo]
                        template: undefined,
                        yahoo: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        }
                    },

                    isCobrand: false,
                    logoBgSize: 'Bgz(250px)',
                    backgroundColor: null,
                    backgroundPosX: 'Bgpx(0)',
                    backgroundPosY: 'Bgpy(0)',
                    backgroundPosition: undefined,
                    height: 74,
                    width: 125,
                    logoLeftWidth: undefined,
                    logoRightWidth: undefined,
                    logoMarginLeft: undefined,
                    logoMarginRight: undefined,
                    defaultRetinaImage:'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage_2x.png',
                    defaultNonRetinaImage:'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage.png',
                    image: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32_2x.png',
                    nonRetinaImage: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32.png',
                    link: undefined,
                    partnerLink: undefined,
                    text: undefined,
                    showYahooOnlyLogo: undefined
                },
                mail: {
                    // mail: style
                    style: {
                        button: {
                            content_1: undefined,
                            content_2: undefined,
                            icon_1: undefined,
                            icon_2: undefined,
                            inline: {
                                WebkitFontSmoothing: undefined
                            },
                            text_1: undefined,
                            text_2: undefined,
                            wrapper: undefined
                        },
                        mailcount: {
                            wrapper_1: undefined
                        }
                    },

                    // mail: data
                    data: {
                        icon: {
                            mail: {
                                name: 'NavMail',
                                size: 24
                            }
                        }
                    },

                    // [start] TODO: migrate to mail:data
                    urls: {
                        mail: {
                            protocol: 'https',
                            hostname: 'mail.yahoo.com',
                            pathname: '/',
                            query: {
                                '.intl': undefined,
                                '.lang': undefined,
                                '.partner': undefined,
                                '.src': undefined
                            }
                        }
                    },
                    compose: {
                        url: 'https://mrd.mail.yahoo.com/compose'
                    },
                    count: {
                        api: {
                            protocol: 'https',
                            host: 'mg.mail.yahoo.com',
                            path: '/mailservices/v1/newmailcount',
                            query: {
                                appid: 'UnivHeader',
                                wssid: ''
                            }
                        },
                        jsonp: {
                            timeout: 3000,
                            prefix: '__uhmc__'
                        },
                        maxCountDisplay: 99,
                        pollingInterval: 2,
                        pollingDuration: 10
                    },
                    mailservices: {
                        api: {
                            host: 'mail.yahoo.com',
                            path: '/',
                            protocol: 'https',
                            query: {
                                '.src': 'ym'
                            }
                        }
                    },
                    preview: {
                        api: {
                            protocol: 'https',
                            host: 'ucs.query.yahoo.com',
                            path: '/v1/console/yql',
                            query: {
                                format: 'json',
                                q: 'select messageInfo.receivedDate, messageInfo.mid, messageInfo.flags.isRead, ' +
                                   'messageInfo.from.name, messageInfo.subject from ymail.messages ' +
                                   'where numMid="3" limit 6',
                                crumb: ''
                            }
                        },
                        fetchInterval: 5000,
                        jsonp: {
                            timeout: 3000,
                            prefix: '__uhmp__'
                        },
                        urls: {
                            message: 'https://mrd.mail.yahoo.com/msg?fid=Inbox&src=hp&mid='
                        }
                    },
                    url: 'https://mail.yahoo.com/',
                    signedInUrl: 'https://mail.yahoo.com/'
                    // [end] TODO: migrate to mail:data
                },
                profile: {
                    // profile: style
                    style: {
                        account_info: {
                            alias: undefined,
                            content: undefined,
                            name: undefined,
                            wrapper: undefined
                        },
                        avatar: {
                            content: undefined,
                            color: undefined,
                            name: undefined,
                            wrapper: undefined
                        },
                        container: {
                            wrapper: undefined
                        },
                        panel: {
                            wrapper: undefined
                        },
                        button: {
                            content_1: undefined,
                            content_2: undefined,
                            inline: {
                                WebkitFontSmoothing: undefined
                            },
                            wrapper: undefined
                        },
                        signed_out: {
                            content: undefined,
                            wrapper: undefined
                        }
                    },

                    // profile: data
                    data: {
                        icon: {
                            profile: {
                                name: 'profile',
                                size: 34
                            }
                        }
                    },
                    // [start] TODO: migrate to profile:data
                    // Load the component in either 'client' or 'server'.
                    mode: 'server',
                    avatarSize: '36px',
                    // TODO: Remove these after code change in ProfileMobile & ProfileCard.
                    // [START]
                    loginBaseUrl: 'https://login.yahoo.com/config/login',
                    settingUrl: 'https://edit.yahoo.com/config/eval_profile',
                    signoutUrl: 'https://login.yahoo.com/config/login?logout=1&.direct=2&' +
                                '.done=https://www.yahoo.com',
                    // [END]
                    urls: {
                        signed_in: {
                            protocol: 'https',
                            hostname: 'login.yahoo.com',
                            pathname: '/config/login',
                            // To override .intl params in query params, please use %5B and %5D for keys contain '.'.
                            // e.g.
                            // http://localhost:3000/console?ctrl=Header&m_id=tdv2-applet-uh&config.uh.profile.urls.signed_in.query%5B.intl%5D=x7xy
                            query: {
                                '.done': undefined,
                                '.intl': undefined,
                                '.lang': undefined,
                                '.partner': undefined,
                                '.src': undefined
                            }
                        },
                        signed_out: {
                            protocol: 'https',
                            hostname: 'login.yahoo.com',
                            pathname: '/config/login',
                            query: {
                                '.direct': 2,
                                '.done': undefined,
                                '.partner': undefined,
                                '.src': undefined,
                                logout: 1
                            }
                        },
                        account_info: {
                            protocol: 'https',
                            hostname: 'edit.yahoo.com',
                            pathname: '/config/eval_profile',
                            query: {
                                '.partner': undefined,
                                '.src': undefined
                            }
                        }
                    }
                    // [end] TODO: migrate to profile:data
                },
                search: {
                    // search: style
                    style: {
                        cancel_button: {
                            content: undefined
                        },
                        container: {
                            content: undefined,
                            wrapper: undefined
                        },
                        input: {
                            content: undefined,
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        search_button: {
                            content_1: undefined,
                            content_2: undefined,
                            inline: {
                                filter: undefined,
                                WebkitFontSmoothing: undefined
                            }
                        }
                    },
                    // [START] Deprecated
                    searchFormWidth: undefined,
                    // [END] Deprecated

                    // search: data
                    action: undefined,
                    autofocus: true,
                    assistYlc: ';_ylc=X3oDMTFiaHBhMnJmBF9TAzIwMjM1MzgwNzUEaXRjAzEEc2VjA3NyY2hfcWEEc2xrA3NyY2hhc3Q-',
                    ylc: ';_ylc=X3oDMTFiN25laTRvBF9TAzIwMjM1MzgwNzUEaXRjAzEEc2VjA3NyY2hfcWEEc2xrA3NyY2h3ZWI-',
                    buttonText: 'SEARCH',
                    glowEnabled: false,
                    placeholderIcon: false,
                    placeholderText: 'SEARCH',
                    autocomplete: {
                        gossip: {
                            url: {
                                host: 'search.yahoo.com',
                                path: '/sugg/gossip/gossip-us-ura/',
                                query: {
                                    appid: 'yahoo.com',
                                    nresults: 10,
                                    output: 'sd1'
                                }
                            }
                        }
                    },
                    instantSearch: false,
                    searchHint: false,
                    cancelBtn: false,
                    clearBtn: false,
                    termCompleteBtn: false,
                    magnifyingGlass: false,
                    queries: {
                        hspart: undefined,
                        hsimp: undefined,
                        type: undefined
                    }
                },
                searchOrigin: {
                    fr: 'yfp-t-{BUCKET}',
                    fr2: undefined
                },
                topbar: {
                    // topbar: style
                    style: {
                        nav_home: {
                            content: undefined,
                            inline: {
                                boxShadow: undefined,
                                border: undefined,
                                borderColor: undefined
                            },
                            link: {
                                content: undefined
                            },
                            wrapper: undefined
                        },
                        nav_main: {
                            content: undefined,
                            link: {
                                content: undefined,
                                wrapper: undefined
                            },
                            link_home: {
                                content_1: undefined,
                                content_2: undefined,
                                icon: {
                                    down_arrow: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginLeft: undefined,
                                            verticalAlign: undefined
                                        }
                                    },
                                    home: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginRight: undefined,
                                            marginTop: undefined,
                                            verticalAlign: undefined
                                        }
                                    }
                                },
                                text: undefined,
                                wrapper_1: undefined,
                                wrapper_2: undefined,
                                wrapper_3: undefined
                            },
                            link_more: {
                                content_1: undefined,
                                content_2: undefined,
                                icon: {
                                    down_arrow: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginLeft: undefined,
                                            fill: undefined,
                                            verticalAlign: undefined
                                        }
                                    }
                                },
                                text: undefined,
                                wrapper_1: undefined,
                                wrapper_2: undefined
                            },
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        nav_more: {
                            content: undefined,
                            inline: {
                                boxShadow: undefined,
                                border: undefined,
                                borderColor: undefined
                            },
                            link: {
                                content: undefined
                            },
                            wrapper: undefined
                        }
                    },

                    // topbar: data
                    data: {
                        // Keep the absolute link.
                        // Don't convert to relative link using hllib.convertUrl().
                        // remark: will only be applied if force_absolute_all=false
                        force_absolute: {
                            answers: false,
                            finance: false,
                            games: false,
                            groups: false,
                            home: false,
                            mail: false,
                            mobile: false,
                            screen: false,
                            search: false,
                            shopping: false,
                            sports: false,
                            weather: false
                        },
                        force_absolute_all: true,
                        more: 12,
                        more_item: {
                            link: 'https://everything.yahoo.com',
                            title: 'More'
                        },
                        icon: {
                            home: {
                                name: 'home',
                                size: 16
                            },
                            down_arrow: {
                                name: 'CoreArrowDown',
                                size: 8
                            }
                        }
                    }
                },
                // uh: data
                data: {
                    action: {
                        load_theme: {
                            cache: {
                                enabled: true
                            }
                        }
                    }
                },
                navTitle: '',
                ui: {
                    back: false,
                    darkTheme: false,
                    follow: false,
                    help: false,
                    logo: true,
                    mail: true,
                    mailcount: true,
                    mailpreview: true,
                    menuBtn: true,
                    openSearchBtn: false,
                    profile: true,
                    profileCardOpen: false,
                    profileCardAvatar: false,
                    profileCardName: false,
                    profileCardEmail: false,
                    profileCardSignout: false,
                    profileCardAcctInfo: false,
                    search: true,
                    share: false,
                    theme: 'default',
                    topbar: false,
                    topbarSticky: true
                },
                useNavTitle: false
            }
        }
    ];
}

module.exports = ycb();
