// ycb/uh/tdv2-applet-uh/tests/mock/components/nav/TopbarItems.jsx
var React = require('react/addons');

var TopbarItems = React.createClass({
    statics: {
        nodes: {
            home: '__getNodeHome',
            more: '__getNodeMore',
            main: '__getNodeMain'
        }
    },
    __getNodeHome: function() {
        var data = null;
        var items = this.state.data.get('home');
        if (items) {
            data = (
                <ul id='topbar-items-home'>
                {
                    items.map(function (item, i) {
                        return (
                            <li key={ item.get('id') } className='home'>{ item.get('id') }</li>
                        );
                    }, this)
                }
                </ul>
            );
        }
        return data;
    },
    __getNodeMore: function() {
        var data = null;
        var items = this.state.data.get('more');
        if (items) {
            data = (
                <ul id='topbar-items-more'>
                {
                    items.map(function (item, i) {
                        return (
                            <li key={ item.get('id') } className='more'>{ item.get('id') }</li>
                        );
                    }, this)
                }
                </ul>
            );
        }
        return data;
    },
    __getNodeMain: function() {
        var data = null;
        var items = this.state.data.get('main');
        if (items) {
            data = (
                <ul id='topbar-items-main'>
                {
                    items.map(function (item, i) {
                        return (
                            <li key={ item.get('id') } className='main'>{ item.get('id') }</li>
                        );
                    }, this)
                }
                </ul>
            );
        }
        return data;
    },
    getInitialState: function() {
        //
        // Using this.props.data in getInitialState for unit-test purpose.
        //  e.g.
        //      <TopbarItems data={ data } />
        //  where data is an immutable object.
        //
        // https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html
        //
        return {
            data: this.props.data
        };
    },
    getNode: function(name) {
        var data = null;
        if (TopbarItems.nodes[name]) {
            data = this[TopbarItems.nodes[name]].call(this);
        }
        return data;
    },
    render: function() {
        return (
            <div id='topbar-items' className='topbar-items'>
                { this.getNode('home') }
                { this.getNode('main') }
                { this.getNode('more') }
            </div>
        );
    }
});

module.exports = TopbarItems;
