// ycb/uh/tdv2-applet-uh/configs/components/device/index.js
function ycb() {
    return [
        {
            settings: ['device:tablet'],
            uh:{
                search:{
                    autofocus: false,
                    style: {
                        input: {
                            wrapper_2: 'W(40vw)'
                        }
                    }
                }
            }
        },
        {
            settings: ['device:smartphone'],
            uh: {
                theme:{
                    followNotificationBorderColor: 'Bdc(#2e4bbe)',
                    mailNotificationBorderColor: 'Bdc(#501e9f)',
                    mobileHeaderBackground: 'Bg(fallbackMobileUHColor) Bg(customGradientUH)!',
                    mobileHeaderText: 'C(#fff)',
                    mobileHeaderIcon: '#fff'
                },
                logo: {
                    backgroundColor: 'transparent',
                    image: 'https://s.yimg.com/rz/uh/yahoo_frontpage_en-US_s_f_w_bestfit_frontpage.png',
                    width: 'W(100px)',
                    height: 'H(25px)',
                    backgroundPosition: 'left 0'
                },
                mail: {
                    count: {
                        pollingDuration: 2
                    }
                },
                navTitle: '',
                search: {
                    autofocus: false,
                    placeholderIcon: true,
                    cancelBtn: true,
                    clearBtn: true,
                    termCompleteBtn: true,
                    magnifyingGlass: true
                },
                useNavTitle: false,
                ui: {
                    back: false,
                    follow: false,
                    logo: true,
                    mail: true,
                    mailcount: true,
                    mailpreview: false,
                    menuBtn: true,
                    openSearchBtn: false,
                    profile: true,
                    profileCardOpen: false,
                    profileCardAvatar: false,
                    profileCardName: true,
                    profileCardEmail: true,
                    profileCardSignout: true,
                    profileCardAcctInfo: true,
                    search: true,
                    share: false,
                    topbar: false
                }
            }
        }
    ];
}

module.exports = ycb();
