// ycb/uh/tdv2-applet-uh/services/UHCrumbsService.js
/**
 * UHCrumbs Service
 * @class CrumbService
 */

'use strict';

var debug = require('debug')('UH:UHCrumbsService');
var api = require('td-api');
var createService = api.base.service;

var UHCrumbs = createService({
    /**
     * Service name
     * @property name
     * @type {String}
     * @required
     */
    name: 'UHCrumbsService',

    /**
     * Get header data
     * @method read
     * @param {Object} req Request object
     * @param {String} resource The resource name
     * @param {Object} params Options for the resource
     * @param {Object} [configs] Per request configuration
     * @param {Function} callback
     * @async
     */
    read: function read(req, resource, params, configs, callback) {
        debug('Read params', params);
        // checking req.xhr to validate that it is an ajax call
        if (req && req.yck && req.xhr === true) {
            req.yck.cookie_init_header(req.headers.cookie);
            var ucsCrumb = req.yck.cookie_issue_crumb('ucscrumb');
            var mailCountCrumb = req.yck.cookie_issue_crumb('wssid');
            var accountSwitchCrumb = req.yck.cookie_issue_crumb('login_auth_crumb');
            callback(null, {
                ucsCrumb: ucsCrumb,
                mailCountCrumb: mailCountCrumb,
                accountSwitchCrumb: accountSwitchCrumb
            });
        } else {
            // Should error code be returned?
            callback(null, '');
        }
    }
});

module.exports = UHCrumbs;
