// ycb/uh/tdv2-applet-uh/tests/mock/config.js
var Immutable = require('immutable');
var data =
{
    'backgroundColor': 'Bgc(#f9f9f9)',
    'theme':{
        'topbarBackground': 'Bgc(#2d1152)',
        'topbarIcon': '#1d1da3',
        'profileBackground': 'Bgc(uhPurple):h C(uhPurple) Bdc(uhPurple)',
        'profileIcon': '#400090',
        'mailBackground': 'Bgc(uhPurple)',
        'mailNotificationBackground': 'Bgc(#f0162f)',
        'mailNotificationBorderColor': 'Bdc(#fff)',
        'followNotificationBorderColor': 'Bdc(#fff)'
    },
    logo: {
        style: {
            cobrand: {
                partner: {
                    content_1: undefined,
                    content_2: undefined
                },
                image_1x: undefined,
                image_2x: undefined,
                wrapper_1: undefined,
                wrapper_2: undefined,
                yahoo: {
                    content_1: undefined,
                    content_2: undefined
                }
            },
            container: {
                inline: {
                    backgroundColor: undefined
                },
                wrapper_1: undefined,
                wrapper_2: undefined
            },
            country_property: {
                content_1: undefined,
                content_2: undefined,
                image_1x: undefined,
                image_2x: undefined
            },
            property: {
                content: undefined,
                image_1x: undefined,
                image_2x: undefined
            },
            // template: [cobrand | property | yahoo]
            template: undefined,
            yahoo: {
                content: undefined,
                image_1x: undefined,
                image_2x: undefined
            }
        },

        logoBgSize: 'Bgz(250px)',
        backgroundColor: null,
        backgroundPosX: 'Bgpx(0)',
        backgroundPosY: 'Bgpy(0)',
        backgroundPosition: undefined,
        height: 74,
        width: 125,
        logoLeftWidth: undefined,
        logoRightWidth: undefined,
        logoMarginLeft: undefined,
        magData: undefined,
        defaultRetinaImage:'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage_2x.png',
        defaultNonRetinaImage:'https://s.yimg.com/rz/d/yahoo_frontpage_en-US_s_f_pw_bestfit_frontpage.png',
        image: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32_2x.png',
        nonRetinaImage: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32.png',

        link: undefined,
        partnerLink: undefined
    },
    mail: {
        compose: {
            url: 'https://mrd.mail.yahoo.com/compose'
        },
        count: {
            api: {
                host: 'mg.mail.yahoo.com',
                path: '/mailservices/v1/newmailcount',
                protocol: 'https',
                query: {
                    appid: 'UnivHeader',
                    wssid: ''
                }
            },
            jsonp: {
                'prefix': '__uhmc__',
                'timeout': 3000
            },
            maxCountDisplay: 99,
            pollingDuration: 10,
            pollingInterval: 2
        },
        mailservices: {
            api: {
                host: 'mail.yahoo.com',
                path: '/',
                protocol: 'https',
                query: {
                    '.src': 'ym'
                }
            }
        },
        preview: {
            api: {
                host: 'ucs.query.yahoo.com',
                path: '/v1/console/yql',
                protocol: 'https',
                query: {
                    'crumb': '',
                    'format': 'json',
                    'q':
                    [
                        'select messageInfo.receivedDate, messageInfo.mid, messageInfo.flags.isRead,',
                        'messageInfo.from.name, messageInfo.subject',
                        'from ymail.messages where numMid=\'3\' limit 6'
                    ].join(' ')
                }
            },
            fetchInterval: 5000,
            jsonp: {
                prefix: '__uhmp__',
                timeout: 3000
            },
            urls: {
                message: 'https://mrd.mail.yahoo.com/msg?fid=Inbox&src=hp&mid='
            }
        },
        url: 'https://mail.yahoo.com/'
    },
    profile: {
        style: {
            button: {
                content_wrapper: 'V(h)'
            },
            signed_in: {
                content: undefined,
                content_wrapper: undefined,
                inline: undefined
            }
        },
        'avatarSize': '36px',
        'loginBaseUrl': 'https://login.yahoo.com/config/login',
        'settingUrl': 'https://edit.yahoo.com/config/eval_profile',
        'signoutUrl': 'https://login.yahoo.com/config/login?logout=1&.direct=2&.done=https://www.yahoo.com',
        'urls': {
            'account_info': {
                'hostname': 'edit.yahoo.com',
                'pathname': '/config/eval_profile',
                'protocol': 'https',
                'query': {}
            },
            'signed_in': {
                'hostname': 'login.yahoo.com',
                'pathname': '/config/login',
                'protocol': 'https',
                'query': {}
            },
            'signed_out': {
                'hostname': 'login.yahoo.com',
                'pathname': '/config/login',
                'protocol': 'https',
                'query': {
                    '.direct': 2,
                    '.done': 'https://www.yahoo.com/',
                    'logout': 1
                }
            }
        }
    },
    search: {
        style: {
            cancel_button: {
                content: undefined
            },
            container: {
                content_wrapper: undefined
            },
            input: {
                content: undefined,
                content_wrapper: undefined,
                form_width: undefined
            },
            search_button: {
                content_1: undefined,
                content_2: undefined,
                inline: {
                    filter: undefined,
                    WebkitFontSmoothing: undefined
                }
            }
        },
        'action': 'https://search.yahoo.com/search',
        'assistYlc': ';_ylc=X3oDMTFiaHBhMnJmBF9TAzIwMjM1MzgwNzUEaXRjAzEEc2VjA3NyY2hfcWEEc2xrA3NyY2hhc3Q-',
        'autocomplete': {
            'gossip': {
                'url': {
                    'host': 'search.yahoo.com',
                    'path': '/sugg/gossip/gossip-us-ura/',
                    'query': {
                        'appid': 'yahoo.com',
                        'nresults': 10,
                        'output': 'sd1'
                    }
                }
            }
        },
        'autofocus': true,
        'cancelBtn': false,
        'glowEnabled': false,
        'clearBtn': false,
        'instantSearch': false,
        'magnifyingGlass': false,
        'placeholder': '',
        'placeholderIcon': false,
        'queries': {
            'fr': ''
        },
        'searchHint': false,
        'termCompleteBtn': false,
        'ylc': ';_ylc=X3oDMTFiN25laTRvBF9TAzIwMjM1MzgwNzUEaXRjAzEEc2VjA3NyY2hfcWEEc2xrA3NyY2h3ZWI-'
    },
    ui: {
        back: false,
        darkTheme: false,
        follow: false,
        logo: true,
        mail: true,
        mailcount: true,
        mailpreview: true,
        profile: true,
        search: true,
        setting: true,
        topbar: true
    },
    topbar: {
        // topbar: style
        style: {
            nav_home: {
                content: undefined,
                inline: {
                    boxShadow: undefined,
                    border: undefined,
                    borderColor: undefined
                },
                link: {
                    content: undefined
                },
                wrapper: undefined
            },
            nav_main: {
                content: undefined,
                link: {
                    content: undefined,
                    wrapper: undefined
                },
                link_home: {
                    content: undefined,
                    content_1: undefined,
                    icon: {
                        down_arrow: {
                            color: undefined,
                            color_h: undefined,
                            inline: {
                                cursor: undefined,
                                marginLeft: undefined,
                                verticalAlign: undefined
                            }
                        },
                        home: {
                            color: undefined,
                            color_h: undefined,
                            inline: {
                                cursor: undefined,
                                marginRight: undefined,
                                marginTop: undefined,
                                verticalAlign: undefined
                            }
                        }
                    },
                    text: undefined,
                    wrapper: undefined,
                    wrapper_1: undefined,
                    wrapper_2: undefined
                },
                link_more: {
                    content: undefined,
                    content_1: undefined,
                    icon: {
                        color: undefined,
                        color_h: undefined,
                        inline: {
                            cursor: undefined,
                            marginLeft: undefined,
                            fill: undefined,
                            verticalAlign: undefined
                        }
                    },
                    text: undefined,
                    wrapper: undefined,
                    wrapper_1: undefined
                },
                wrapper: undefined,
                wrapper_1: undefined
            },
            nav_more: {
                content: undefined,
                inline: {
                    boxShadow: undefined,
                    border: undefined,
                    borderColor: undefined
                },
                link: {
                    content: undefined
                },
                wrapper: undefined
            }
        },

        // topbar: data
        data: {
            force_absolute: {
                answers: true,
                home: true,
                finance: true,
                games: true,
                groups: true,
                mail: true,
                mobile: true,
                screen: true,
                search: true,
                shopping: true,
                sports: true,
                weather: true
            },
            force_absolute_all: false,
            icon: {
                home: {
                    name: 'home',
                    size: 16
                },
                down_arrow: {
                    name: 'CoreArrowDown',
                    size: 8
                }
            },
            more: 10,
            more_item: {
                link: 'https://everything.yahoo.com/',
                title: 'More'
            }
        }
    },
    data: {
        action: {
            load_theme: {
                cache: {
                    enabled: true
                }
            }
        }
    }
};

module.exports = Immutable.fromJS(data);
