// ycb/uh/tdv2-applet-uh/tests/mock/Hello.jsx
var _require = require('root-require');
var Immutable = require('immutable');
var React = require('react/addons');
var AppletMixin = _require('mixins/AppletMixin');

var CONSTANTS = Immutable.fromJS({
    'i13n': {
        'signed_in': {
            elm: 'btn',
            itc: '0',
            slk: 'signin',
            subsec: 'profile'
        },
        'signed_out': {
            elm: 'btn',
            itc: '0',
            slk: 'signout',
            subsec: 'profile'
        }
    }
});

var Hello = React.createClass({

    contextTypes: {
        conf: React.PropTypes.object.isRequired,
        session: React.PropTypes.object.isRequired,
        theme: React.PropTypes.object.isRequired
    },

    mixins: [AppletMixin],

    statics: {
        base: ['profile'],
        className: {
            style: {
                button: {
                    custom: '_getClassCustom'
                }
            }
        },
        conf: {
            style: {
                button: {
                    custom: '_getConfCustom'
                }
            }
        },
        i13n: CONSTANTS.get('i13n'),
        metaData: {
            email: '_getMetaDataEmail'
        }
    },

    _getClassCustom: function() {
        return 'V(v)';
    },

    _getConfCustom: function() {
        return 'V(v)';
    },

    _getMetaDataEmail: function() {
        return 'stanleyn@yahoo-inc.com';
    },

    render: function() {
        return (
            <div>
                <div id='get_class_name'>
                    <div className='conf'>{ this.getClassName(Hello, ['style', 'button', 'content_wrapper']) }</div>
                    <div className='statics'>{ this.getClassName(Hello, ['style', 'button', 'custom']) }</div>
                    <div className='theme'>{ this.getClassName(Hello, ['style', 'button', 'content_1']) }</div>
                </div>
                <div id='get_conf'>
                    <div className='conf'>{ this.getConf(Hello, ['style', 'button', 'content_wrapper']) }</div>
                    <div className='statics'>{ this.getConf(Hello, ['style', 'button', 'custom']) }</div>
                </div>
                <div id='get_meta_data'>
                    <div className='props'>{ this.getMetaData(Hello, 'name') }</div>
                    <div className='statics'>{ this.getMetaData(Hello, 'email') }</div>
                </div>
                <div id='get_i13n'>
                    <div className='signed_in'>{ JSON.stringify(this.getI13n(Hello, 'signed_in')) }</div>
                    <div className='signed_out'>{ JSON.stringify(this.getI13n(Hello, 'signed_out', {itc: 1 + '', id: 'stanleyn'})) }</div>
                </div>
                <div id='get_intl'>
                    <div className='hello_world'>{ this.getIntl('HELLO_WORLD') }</div>
                </div>
                <div id='get_theme'>
                    <div className='content_wrapper'>{ this.getTheme(Hello, ['style', 'button', 'content_wrapper']) }</div>
                </div>
            </div>
        );
    }
});

module.exports = Hello;
