// ycb/uh/tdv2-applet-uh/tests/mock/theme.js
var Immutable = require('immutable');
var data =
{
    style: {
        background: undefined
    },
    logo: {
        style: {
            cobrand: {
                partner: {},
                wrapper_1: 'Bgr(nr) D(b) Fl(start) Mstart(10px)',
                yahoo: {}
            },
            container: {
                inline: {
                    backgroundColor: 'transparent'
                },
                wrapper_1: 'Fl(start) Fz(0) M(0) P(0)',
                wrapper_2: 'Miw(190px) UHMR1D_Py(0px)'
            },
            country_property: {
                content_1: 'Bgpx(0) Bgr(nr) Cur(p) D(b) H(35px)',
                content_2: 'Bgz(702px) Mx(a)! W(155px)'
            },
            property: {
                content: 'Bgr(nr) Bgpx(0) Bgz(darkThemeSize) Cur(p) D(b) H(35px) Mx(a)! W(92px)'
            },
            template: 'yahoo',
            yahoo: {
                content: 'Bgpx(0) Bgr(nr) Bgz(250px) Cur(p) D(ib) Fl(start) H(28px) Mstart(20px) W(125px)',
                image_1x: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32.png',
                image_2x: 'https://s.yimg.com/rz/l/yahoo_en-US_f_pw_125x32_2x.png'
            }
        }
    },
    profile: {
        style: {
            account_info: {
                alias: 'C(textColor) Ell Maw(160px)',
                content: 'C(#0078ff) C(#0078ff):h C(#0078ff):v D(ib) Fz(13px) My(2px)',
                content_wrapper: 'P(16px)',
                name: 'C(textColor) Ell Fw(b) Fz(13px) Maw(160px)'
            },
            avatar: {
                content_wrapper: 'C(#7d00fb) D(ib) Ell H(34px) Maw(120px)',
                content: 'Bdrs(100px) Bgz(cv) D(ib) H(34px) W(34px)',
                color: '#400090',
                name: 'D(n) Lh(2.6) Mstart(4px)'
            },
            container: {
                content_wrapper: 'Fl(start) H(34px) Mx(8px)'
            },
            panel: {
                content_wrapper: [
                    'Bdrs(4px) Bgc(menuBgc) Bxsh(customShadowFlyoutMenu) D(n) End(0) Pos(a)',
                    'T(40px) Ta(start) uh-menu-open_D(b) W(280px)'
                ].join(' ')
            },
            button: {
                content: [
                    'Bdrs(4px) Bds(s) Bdw(1px) C(#fff):h D(ib) Fil(n)',
                    'Maw(120px) Px(26px) Py(8px) Td(n):h'
                ].join(' '),
                content_1: 'Bdc(uhPurple) Bgc(uhPurple):h C(uhPurple)',
                content_wrapper: 'Fl(start) Mx(4px)',
                inline: {
                    WebkitFontSmoothing: undefined
                }
            },
            signed_out: {
                content: 'C(textColor)',
                content_wrapper: 'Bdtc(#e1e1e5) Bdts(s) Bdtw(1px) Mx(16px) Py(16px) Ta(c)'
            }
        }
    }
};

module.exports = Immutable.fromJS(data);
