// ycb/uh/tdv2-applet-uh/configs/services/theme/master.js
function ycb() {
    return [
        {
            settings: ['master'],
            uh: {
                style: {
                    container: {
                        content: undefined,
                        right_list: undefined,
                        wrapper_1: undefined,
                        wrapper_2: undefined,
                        wrapper_3: undefined,
                        wrapper_4: undefined,
                        wrapper_5: undefined,
                        wrapper_6: undefined,
                        wrapper_7: undefined,
                        wrapper_8: undefined
                    }
                },
                // child components
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: undefined,
                                content_2: undefined
                            },
                            image_1x: undefined,
                            image_2x: undefined,
                            wrapper_1: undefined,
                            wrapper_2: undefined,
                            yahoo: {
                                content_1: undefined,
                                content_2: undefined
                            }
                        },
                        container: {
                            inline: {
                                backgroundColor: undefined
                            },
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        country_property: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        property: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        },
                        // template: [cobrand | country_property | property | yahoo]
                        template: undefined,
                        yahoo: {
                            content_1: undefined,
                            content_2: undefined,
                            image_1x: undefined,
                            image_2x: undefined
                        }
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_1: undefined,
                            content_2: undefined,
                            icon_1: undefined,
                            icon_2: undefined,
                            inline: {
                                WebkitFontSmoothing: undefined
                            },
                            text_1: undefined,
                            text_2: undefined,
                            wrapper: undefined
                        },
                        mailcount: {
                            wrapper_1: undefined
                        }
                    }
                },
                profile: {
                    style: {
                        account_info: {
                            alias: undefined,
                            content: undefined,
                            name: undefined,
                            wrapper: undefined
                        },
                        avatar: {
                            content: undefined,
                            color: undefined,
                            name: undefined,
                            wrapper: undefined
                        },
                        container: {
                            wrapper: undefined
                        },
                        panel: {
                            wrapper: undefined
                        },
                        button: {
                            content_1: undefined,
                            content_2: undefined,
                            inline: {
                                WebkitFontSmoothing: undefined
                            },
                            wrapper: undefined
                        },
                        signed_out: {
                            content: undefined,
                            wrapper: undefined
                        }
                    }
                },
                search: {
                    style: {
                        cancel_button: {
                            content: undefined
                        },
                        container: {
                            content: undefined,
                            wrapper: undefined
                        },
                        input: {
                            content: undefined,
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        search_button: {
                            content_1: undefined,
                            content_2: undefined,
                            inline: {
                                filter: undefined,
                                WebkitFontSmoothing: undefined
                            }
                        }
                    }
                },
                topbar: {
                    style: {
                        nav_home: {
                            content: undefined,
                            inline: {
                                boxShadow: undefined,
                                border: undefined,
                                borderColor: undefined
                            },
                            link: {
                                content: undefined
                            },
                            wrapper: undefined
                        },
                        nav_main: {
                            content: undefined,
                            link: {
                                content: undefined,
                                wrapper: undefined
                            },
                            link_home: {
                                content_1: undefined,
                                content_2: undefined,
                                icon: {
                                    down_arrow: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginLeft: undefined,
                                            verticalAlign: undefined
                                        }
                                    },
                                    home: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginRight: undefined,
                                            marginTop: undefined,
                                            verticalAlign: undefined
                                        }
                                    }
                                },
                                text: undefined,
                                wrapper_1: undefined,
                                wrapper_2: undefined,
                                wrapper_3: undefined
                            },
                            link_more: {
                                content_1: undefined,
                                content_2: undefined,
                                icon: {
                                    down_arrow: {
                                        color_1: undefined,
                                        color_2: undefined,
                                        inline: {
                                            cursor: undefined,
                                            marginLeft: undefined,
                                            fill: undefined,
                                            verticalAlign: undefined
                                        }
                                    }
                                },
                                text: undefined,
                                wrapper_1: undefined,
                                wrapper_2: undefined
                            },
                            wrapper_1: undefined,
                            wrapper_2: undefined
                        },
                        nav_more: {
                            content: undefined,
                            inline: {
                                boxShadow: undefined,
                                border: undefined,
                                borderColor: undefined
                            },
                            link: {
                                content: undefined
                            },
                            wrapper: undefined
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
