// ycb/uh/tdv2-applet-uh/configs/components/site/index.js
function ycb() {
    return [
        {
            settings: ['site:autos'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magautos_web_gs',
                    fr2: 'p:auto,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:beauty'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magbeauty_web_gs',
                    fr2: 'p:bty,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:celebrity'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_omg_web_gs_1',
                    fr2: 'p:clb,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:finance'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_finance_web'
                }
            }
        },
        {
            settings: ['site:food'],
            uh: {
                searchOrigin: {
                    fr: 'yshineheader_test2',
                    fr2: 'p:fd,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:health'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_maghealth_web_gs',
                    fr2: 'p:hlth,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:katiecouric'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_news_web_gs',
                    fr2: 'p:news,m:{FRSOURCE}'
                },
                logo: {
                    link: 'https://www.yahoo.com/news'
                }
            }
        },
        {
            settings: ['site:makers'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magdiy_web_gs',
                    fr2: 'p:mk,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:movies'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magmovies_web_gs',
                    fr2: 'p:mv,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:music'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_music_web_gs',
                    fr2: 'p:msc,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:news'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_news_web_gs',
                    fr2: 'p:news,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:omg'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_omg_web_gs',
                    fr2: 'p:clb,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:parenting'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magparenting_web_gs',
                    fr2: 'p:prnt,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:politics'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_news_web_gs_t2',
                    fr2: 'p:pol,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:sports'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_sports_web_gs',
                    fr2: 'p:sprt,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:sports', 'lang:en-US'],
            uh: {
                logo: {
                    link: 'http://sports.yahoo.com/'
                }
            }
        },
        {
            settings: ['site:sports', 'lang:zh-Hant-TW'],
            uh: {
                logo: {
                    link: 'http://tw.sports.yahoo.com/'
                }
            }
        },
        {
            settings: ['site:style'],
            uh: {
                searchOrigin: {
                    fr: 'magstyle_web_gs',
                    fr2: 'p:stl,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:tech'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magtech_web_gs',
                    fr2: 'p:tech,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:travel'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magtravel_web_gs',
                    fr2: 'p:trvl,m:{FRSOURCE}'
                }
            }
        },
        {
            settings: ['site:tv'],
            uh: {
                searchOrigin: {
                    fr: 'uh3_magtv_web_gs',
                    fr2: 'p:tv,m:{FRSOURCE}'
                }
            }
        }
    ];
}

module.exports = ycb();
