// ycb/uh/tdv2-applet-uh/bundle.js
module.exports = {
    controllerViews: {
        AttHeader: require('./components-dist/Header'), // backward compatible
        AttMobileHeader: require('./components-dist/MobileHeader'), // backward compatible
        Header: require('./components-dist/Header'),
        MobileHeader: require('./components-dist/MobileHeader'),
        ProfileCard: require('./components-dist/ProfileCard'),
        TabletHeader: require('./components-dist/TabletHeader'),
        TumblrHeader: require('./components-dist/Header') // backward compatible
    },
    stores: {
        HeaderStore: require('./stores/HeaderStore'),
        MailStore: require('./stores/MailStore'),
        MobileHeaderStore: require('./stores/MobileHeaderStore'), // for mobile topic title
        NavStore: require('./stores/NavStore'),
        UHCrumbsStore: require('./stores/UHCrumbsStore'),
        // ProfileStore: require('tdv2-service-profile').ProfileStore,
        // SearchInputStore: require('tdv2-applet-search-input').stores.SearchInputStore,
        ThemeStore: require('./stores/ThemeStore')
    }
};
