// ycb/uh/tdv2-applet-uh/configs/services/theme/partner/att.js
function ycb() {
    return [
        {
            settings: ['partner:att', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: 'Cur(p) D(ib) H(31px) Pos(a)',
                                content_2: 'W(95px)'
                            },
                            image_1x: 'https://s.yimg.com/rz/d/att_en-US_f_pw_351x40.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_en-US_f_pw_351x40_2x.png',
                            wrapper_2: 'Bgz(702px) H(31px) W(170px) Bgpx(0) Bgpy(0)',
                            yahoo: {
                                content_1: 'Cur(p) D(ib) H(31px) Pos(a) W(75px)',
                                content_2: 'Mstart(95px)!'
                            }
                        },
                        container: {
                            wrapper_2: 'Miw(190px) Py(4px) UHMR1D_Pb(0px)'
                        },
                        template: 'cobrand'
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_2: 'Bgc(#067ab4)'
                        }
                    }
                },
                profile: {
                    style: {
                        avatar: {
                            color: '#067ab4'
                        },
                        button: {
                            content_2: 'Bdc(#067ab4) Bgc(#067ab4):h C(#067ab4)'
                        }
                    }
                },
                topbar: {
                    style: {
                        nav_main: {
                            link_home: {
                                icon: {
                                    down_arrow: {
                                        color_2: '#067ab4'
                                    },
                                    home: {
                                        color_2: '#067ab4'
                                    }
                                },
                                wrapper_3: 'Pend(6px)'
                            },
                            link_more: {
                                icon: {
                                    down_arrow: {
                                        color_2: '#067ab4'
                                    }
                                }
                            },
                            wrapper_2: 'Bgc(#067ab4)'
                        }
                    }
                }
            }
        },
        {
            settings: ['partner:att', 'lang:en-US', 'theme:dark'],
            uh: {
                profile: {
                    style: {
                        avatar: {
                            color: '#fff'
                        },
                        button: {
                            content_2: [
                                'DarkTheme_Bdc(darkthemeColor) DarkTheme_Bdc(#fff):h',
                                'DarkTheme_Bgc(darkthemeColor) DarkTheme_Bgc(#fff):h DarkTheme_C(#000):h C(#fff)'
                            ].join(' ')
                        }
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_2: 'Bgc(darkthemeColor) Bgc(#fff):h'
                        }
                    }
                }
            }
        },
        {
            settings: ['partner:att', 'lang:es-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: 'D(ib) Cur(p) H(31px) Pos(a)',
                                content_2: 'W(95px)'
                            },
                            image_1x: 'https://s.yimg.com/rz/d/att_es-US_f_pw_351x40.png',
                            image_2x: 'https://s.yimg.com/rz/d/att_es-US_f_pw_351x40_2x.png',
                            wrapper_2: 'Bgz(702px) H(31px) W(170px) Bgpx(0) Bgpy(0)',
                            yahoo: {
                                content_1: 'D(ib) Cur(p) H(31px) Pos(a) W(75px)',
                                content_2: 'Mstart(95px)!'
                            }
                        },
                        container: {
                            wrapper_2: 'Miw(190px) Py(4px) UHMR1D_Pb(0px)'
                        },
                        template: 'cobrand'
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_2: 'Bgc(#067ab4)'
                        }
                    }
                },
                profile: {
                    style: {
                        avatar: {
                            color: '#067ab4'
                        },
                        button: {
                            content_2: 'Bdc(#067ab4) Bgc(#067ab4):h C(#fff)'
                        }
                    }
                },
                topbar: {
                    style: {
                        nav_main: {
                            link_home: {
                                icon: {
                                    down_arrow: {
                                        color_2: '#067ab4'
                                    },
                                    home: {
                                        color_2: '#067ab4'
                                    }
                                },
                                wrapper_3: 'Pend(6px)'
                            },
                            link_more: {
                                icon: {
                                    down_arrow: {
                                        color_2: '#067ab4'
                                    }
                                }
                            },
                            wrapper_2: 'Bgc(#067ab4)'
                        }
                    }
                }
            }
        },
        {
            settings: ['partner:att', 'lang:es-US', 'theme:dark'],
            uh: {
                profile: {
                    style: {
                        avatar: {
                            color: '#fff'
                        },
                        button: {
                            content_2: [
                                'DarkTheme_Bdc(darkthemeColor) DarkTheme_Bdc(#fff):h',
                                'DarkTheme_Bgc(darkthemeColor) DarkTheme_Bgc(#fff):h DarkTheme_C(#000):h C(#fff)'
                            ].join(' ')
                        }
                    }
                },
                mail: {
                    style: {
                        button: {
                            content_2: 'Bgc(darkthemeColor) Bgc(#fff):h'
                        }
                    }
                }
            }
        },
    ];
}

module.exports = ycb();
