// ycb/uh/tdv2-applet-uh/configs/services/theme/partner/frontier.js
function ycb() {
    return [
        {
            settings: ['partner:frontier', 'lang:en-US'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: 'Cur(p) D(ib) H(31px) Pos(a)',
                                content_2: 'W(85px)'
                            },
                            image_1x: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_351x40.png',
                            image_2x: 'https://s.yimg.com/rz/d/frontier_en-US_f_pw_351x40_2x.png',
                            wrapper_2: 'Bgz(702px) H(31px) W(170px) Bgpx(0) Bgpy(0)',
                            yahoo: {
                                content_1: 'Cur(p) D(ib) H(31px) Pos(a) W(85px)',
                                content_2: 'Mstart(85px)!'
                            }
                        },
                        container: {
                            wrapper_2: 'Miw(190px) Py(4px) UHMR1D_Pb(0px)'
                        },
                        template: 'cobrand'
                    }
                },
                topbar: {
                    style: {
                        nav_main: {
                            link_home: {
                                wrapper_3: 'Pend(6px)'
                            }
                        }
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
