// ycb/uh/tdv2-applet-uh/configs/services/theme/partner/rogers.js
function ycb() {
    return [
        {
            settings: ['partner:rogers', 'lang:en-CA'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: 'Cur(p) D(ib) H(28px) Pos(a)',
                                content_2: 'W(90px)'
                            },
                            image_1x: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40.png',
                            image_2x: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40_2x.png',
                            wrapper_2: 'Bgz(702px) H(28px) W(170px) Bgpx(0) Bgpy(0)',
                            yahoo: {
                                content_1: 'Cur(p) D(ib) H(28px) Pos(a) W(80px)',
                                content_2: 'Mstart(90px)!'
                            }
                        },
                        container: {
                            wrapper_2: 'Miw(190px) Py(4px) UHMR1D_Pb(0px)'
                        },
                        template: 'cobrand'
                    }
                }
            }
        },
        {
            settings: ['partner:rogers', 'lang:fr-CA'],
            uh: {
                logo: {
                    style: {
                        cobrand: {
                            partner: {
                                content_1: 'Cur(p) D(ib) H(25px) Pos(a)',
                                content_2: 'W(90px)'
                            },
                            image_1x: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40.png',
                            image_2x: 'https://s.yimg.com/rz/d/rogers_en-CA_f_pw_351x40_2x.png',
                            wrapper_2: 'Bgz(702px) H(25px) W(170px) Bgpx(0) Bgpy(0)',
                            yahoo: {
                                content_1: 'Cur(p) D(ib) H(25px) Pos(a) W(80px)',
                                content_2: 'Mstart(90px)!'
                            }
                        },
                        container: {
                            wrapper_2: 'Miw(190px) Py(4px) UHMR1D_Pb(0px)'
                        },
                        template: 'cobrand'
                    }
                }
            }
        }
    ];
}

module.exports = ycb();
